From 1e4992a6e4effc3103308fdbe9254fef8bd9b74b Mon Sep 17 00:00:00 2001
From: Balaji Selvanathan <quic_bselvana@quicinc.com>
Date: Thu, 6 Mar 2025 11:43:13 +0530
Subject: [PATCH] PENDING: Add a weak extra fixup_dt_handler

Added a weak extra fixup_dt_handler called fixup_dt_handler_extra in
ft_board_setup in arch/arm/mach-snapdragon/of_fixup.c; users can do
extra dt fixup operations using this function

Signed-off-by: Balaji Selvanathan <quic_bselvana@quicinc.com>
Upstream-Status: Pending
---
 arch/arm/mach-snapdragon/of_fixup.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/mach-snapdragon/of_fixup.c b/arch/arm/mach-snapdragon/of_fixup.c
index bcbd211bbb1..565fc1a549f 100644
--- a/arch/arm/mach-snapdragon/of_fixup.c
+++ b/arch/arm/mach-snapdragon/of_fixup.c
@@ -161,12 +161,18 @@ __weak void fixup_dt_handler(struct fdt_header *fdt_ptr)
 {
 }
 
+__weak void fixup_dt_handler_extra(struct fdt_header *fdt_ptr)
+{
+}
+
 int ft_board_setup(void *blob, struct bd_info __maybe_unused *bd)
 {
 	struct fdt_header *fdt = blob;
 	int node;
 
 	fixup_dt_handler(fdt);
+	fixup_dt_handler_extra(fdt);
+
 	/* We only want to do this fix-up for the RB1 board, quick return for all others */
 	if (!fdt_node_check_compatible(fdt, 0, "qcom,qrb4210-rb2"))
 		return 0;
-- 
2.34.1

